### PR TITLE
Logging e2e tests names in Kusto

### LINKED ISSUE #5574: Real Service Tests - Add e2e test name/info to logs
## Work Item
**Describe the outcome you expect**

Looking at logs from a particular run of e2e tests, we should be able to easily track back to the test that was running to generate those logs.

**Approach**

Mocha seems to support getting the fully "namespaced" test case name (including all the nested `describe` declarations, that sounds cool.  From some brief auditing of different e2e tests it seems that usually a new loader is created per test case, so we just need to pass this value through.  And if there are situations where the loader lives over several test cases, then whatever `describe` block matches (and if that doesn't line up, figure something out)

**Open questions**

Any other info besides fully qualified test case name?  Maybe test file would make it easier to find the spot in code?

Maybe use the `hostScenarioName` field here?  Do we also want to populate `hostScenarioStartTime` field?

### PR DESCRIPTION
**Pull request by chensixx**
Fixes #5574

### CODE REVIEW

File: packages/test/mocha-test-setup/mocharc-common.js [hunk 1]
```diff
@@ -14,8 +14,8 @@ function getFluidTestMochaConfig(packageDir, additionalRequiredModules) {
     const moduleDir = `${packageDir}/node_modules`;
 
     const requiredModules = [
+        ...(additionalRequiredModules ? additionalRequiredModules : []),
         `@fluidframework/mocha-test-setup`, // General mocha setup e.g. suppresses console.log
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/mocharc-common.js:18]: can you add comments about why it is important the mocha-test-setup is last in required modules

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 2]
```diff
@@ -4,28 +4,51 @@
...
-const nullLogger: ITelemetryBufferedLogger = { send: () => {}, flush: async () => {} };
+class TestLogger implements ITelemetryBufferedLogger {
+    send(event: ITelemetryBaseEvent) {
+        event.testName = this.testName;
+        this.parentLogger.send(event);
+    }
+    async flush() {
+        return this.parentLogger.flush();
+    }
+    pass(title: string | undefined) {
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:19]: can you give this method a more descriptive name

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 3]
```diff
@@ -4,28 +4,51 @@
...
 const log = console.log;
 const error = console.log;
+let testLogger: TestLogger;
 export const mochaHooks = {
     beforeAll() {
-        // Ensure getTestLogger is defined
-        if (_global.getTestLogger?.() === undefined) {
-            _global.getTestLogger = () => nullLogger;
-        }
+        const parentLogger = _global.getTestLogger ?? nullLogger;
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:35]: this looks wrong, you need to call the function

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 4]
```diff
@@ -4,28 +4,51 @@
...
+        testLogger = new TestLogger(parentLogger, "");
+        console.log(parentLogger,"parent logger");
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:37]: remove console logs

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 5]
```diff
@@ -4,28 +4,51 @@
...
+        this.parentLogger.send(event);
+    }
+    async flush() {
+        return this.parentLogger.flush();
+    }
+    pass(title: string | undefined) {
+        this.testName = title;
+    }
+    constructor(private readonly parentLogger: ITelemetryBufferedLogger,
+        private testName: string | undefined) {}
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:23]: make test name optional
```suggestion
        private testName?: string) {}
```

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 6]
```diff
@@ -4,28 +4,51 @@
...
 
 import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { Context } from "mocha";
 
 const _global: any = global;
-const nullLogger: ITelemetryBufferedLogger = { send: () => {}, flush: async () => {} };
+class TestLogger implements ITelemetryBufferedLogger {
+    send(event: ITelemetryBaseEvent) {
+        event.testName = this.testName;
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:13]: only add the property if this.testName !== undefined

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 7]
```diff
@@ -4,28 +4,51 @@
...
+        console.log(parentLogger,"parent logger");
+        _global.getTestLogger = () => testLogger;
     },
     beforeEach() {
         // Suppress console.log if not verbose mode
-        if (process.env.FLUID_TEST_VERBOSE === undefined) {
-            console.log = () => { };
-            console.error = () => { };
-        }
+        // if (process.env.FLUID_TEST_VERBOSE === undefined) {
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:42]: un-comment out this code

File: packages/test/mocha-test-setup/src/mochaHooks.ts [hunk 8]
```diff
@@ -4,28 +4,52 @@
...
 const error = console.log;
+let testLogger: TestLogger;
 export const mochaHooks = {
     beforeAll() {
-        // Ensure getTestLogger is defined
-        if (_global.getTestLogger?.() === undefined) {
-            _global.getTestLogger = () => nullLogger;
-        }
+        const parentLogger = _global.getTestLogger?.() ?? nullLogger;
+        testLogger = new TestLogger(parentLogger, "");
```
Comment (anthony-murphy) [inline packages/test/mocha-test-setup/src/mochaHooks.ts:38]:
```suggestion
        testLogger = new TestLogger(parentLogger);
```

**Review by anthony-murphy — APPROVED** [review]

**Comment by markfields** [comment]
Cool @chensixx !

**Pull request merged**